packages/sample/src/main.ts

```diff
@@ -109,33 +121,147 @@ program
+            );
+        }
+        return openExpansionHubWithSerialNumber(
+            serialNumber,
+            parentAddress,
+            moduleAddress,
+        );
+    } else if (parentAddress !== undefined) {
+        return openExpansionHubWithAddress(parentAddress, moduleAddress);
+    }
+
+    let connectedHubs: ParentExpansionHub[] = await openConnectedExpansionHubs();
+    if (connectedHubs.length == 0) {
+        throw new Error("No hubs are connected");
+    }
+    if (connectedHubs.length > 1 || connectedHubs[0].children.length > 0) {
+        throw new Error(
+            "Multiple hubs connected. You must specify a serialNumber and/or address.",
+        );
+    }
+
+    let closeHubs = () => {
+        for (let hub of connectedHubs) {
+            hub.close();
+        }
+    };
+    return [connectedHubs[0], closeHubs];
+}
+
+async function openExpansionHubWithSerialNumber(
+    serialNumber: string,
+    parentAddress: number,
+    moduleAddress: number | undefined,
+): Promise<[hub: ExpansionHub, close: () => void]> {
+    let parentHub = await openParentExpansionHub(serialNumber, parentAddress);
+    if (moduleAddress === undefined || moduleAddress == parentHub.moduleAddress) {
+        return [
+            parentHub,
+            () => {
+                parentHub.close();
+            },
+        ];
+    } else {
+        let childHub = await parentHub.addChildByAddress(moduleAddress);
+        if (childHub.isExpansionHub()) {
+            let closeChild = () => {
+                parentHub.close();
+                if (childHub.isExpansionHub()) {
+                    childHub.close();
+                }
+            };
+            return [childHub, closeChild];
+        } else {
+            throw new Error(
+                `Hub (${serialNumber}) ${moduleAddress} is not an Expansion hub`,
+            );
+        }
+    }
+}
+
+async function openExpansionHubWithAddress(
+    parentAddress: number,
+    moduleAddress: number | undefined,
+): Promise<[hub: ExpansionHub, close: () => void]> {
+    //parent address specified, but no serial number
+    //if the user specifies a module address, use that, else use parent address as module address.
+    let realModuleAddress = moduleAddress !== undefined ? moduleAddress : parentAddress;
+    let serialNumbers = await getPossibleExpansionHubSerialNumbers();
+
+    if (serialNumbers.length > 1) {
+        //there are multiple hubs connected. We can't distinguish without a serial number
+        throw new Error(
+            `There are ${serialNumbers.length} parent hubs. Please specify a serialNumber`,
+        );
+    }
+
+    let [parent, hub] = await openHubWithAddress(
+        serialNumbers[0],
+        parentAddress,
+        realModuleAddress,
+    );
+
+    if (hub.isExpansionHub()) {
+        let closeHub = () => {
+            if (parent.isExpansionHub()) {
+                parent.close();
+            }
+            if (hub.isExpansionHub()) {
```

Comment:
it looks like expansion hubs will automatically close children, so we may not need this.

Reply:
Yeah I'd get rid of it.